src/EditorFeatures/Core/GoToDefinition/AbstractGoToDefinitionSymbolService.cs

```diff
@@ -12,23 +13,36 @@ internal abstract class AbstractGoToDefinitionSymbolService : IGoToDefinitionSym
     {
         protected abstract ISymbol FindRelatedExplicitlyDeclaredSymbol(ISymbol symbol, Compilation compilation);
 
-        public async Task<(ISymbol, TextSpan)> GetSymbolAndBoundSpanAsync(Document document, int position, CancellationToken cancellationToken)
+        public async Task<(ISymbol, TextSpan)> GetSymbolAndBoundSpanAsync(Document document, int position, bool includeLiterals, CancellationToken cancellationToken)
         {
             var workspace = document.Project.Solution.Workspace;
 
             var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             var semanticInfo = await SymbolFinder.GetSemanticInfoAtPositionAsync(semanticModel, position, workspace, cancellationToken).ConfigureAwait(false);
+            var symbol = GetSymbol(semanticInfo, includeLiterals);
 
-            // prefer references to declarations.  It's more likely that the user is attempting to 
-            // go to a definition at some other location, rather than the definition they're on.  
+            if (symbol is null)
+            {
+                return (null, semanticInfo.Span);
+            }
+
+            return (FindRelatedExplicitlyDeclaredSymbol(symbol, semanticModel.Compilation), semanticInfo.Span);
+        }
+
+        private ISymbol GetSymbol(TokenSemanticInfo semanticInfo, bool includeLiterals)
+        {
+            // Prefer references to declarations. It's more likely that the user is attempting to 
+            // go to a definition at some other location, rather than the definition they're on. 
             // This can happen when a token is at a location that is both a reference and a definition.
             // For example, on an anonymous type member declaration.
-            var symbol = semanticInfo.AliasSymbol ??
-                         semanticInfo.ReferencedSymbols.FirstOrDefault() ??
-                         semanticInfo.DeclaredSymbol ??
-                         semanticInfo.Type;
 
-            return (FindRelatedExplicitlyDeclaredSymbol(symbol, semanticModel.Compilation), semanticInfo.Span);
+            var symbol = semanticInfo.AliasSymbol
+                ?? semanticInfo.ReferencedSymbols.FirstOrDefault()
+                ?? semanticInfo.DeclaredSymbol;
+
+            return includeLiterals
+                ? (symbol ?? semanticInfo.Type)
+                : symbol;
```

Comment:
i don't see any check that is seeing we're on a literal or not.

Reply:
Literals don't have an `AliasSymbol`, `ReferencedSymbols`, or `DeclaredSymbol`. Their type for navigation is determined through `Type`.

Reply:
My assumption was: in case there is no alias, referenced or declared symbol it is a literal

Reply:
This means no navigation on anything that doesn't have a symbol, but does have a type.  I would prefer we actually just exclude literals (instread of trying back-infer that we were on a literal).

Reply:
I would recommend instead doing: 

```c#
var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
var isLiteral = syntaxFacts.IsLiteral(whateverTokenYoureOn);
```

That would be an accurate check.  Thanks!

Reply:
You can do this in the code that creates the TokenSemanticInfo.  And the TokenSemanticInfo can include that as a boolean in it.  Or, alternatively, the TokenSemanticInfo can just contain hte token, and you can check it at the call site.  Either way is fine with me.

Reply:
Perfect, thanks for the comments 👍

Reply:
After some investigation, I have some doubts: token "literality" is it semantic or syntactic information? In case you agree that is syntactic info, what do you think about adding `struct TokenSyntacticInfo` and providing methods to populate it, etc? Then I can use it for this case and we can easily extend it with more properties in the future

Reply:
> Token "literality" is it a semantic or syntactic information? 

Currently, it's somethign that can be determined entirely syntactically.

>  In case you agree that is syntactic info, what do you think about adding struct TokenSyntacticInfo

I think that's overkill.  TokenSemanticInfo is really just a struct that means "collect a bunch of useful information about this token for me."  It doesn't mean "it can't contain anything syntactic in it".

> Then I can use it in this PR and we can easily extend it with more properties if future tasks require them

Too speculative :)    Right now we really don't need that.  All we need to know is what token we're ok.  Then ISyntaxFactsService can be trivially used to ask "Is this a literal".